### ISSUE TITLE
Incorrect escape code handling if placed immediately after a link

### ISSUE THREAD

**Issue by sidju** [xtermjs/xterm.js#5229]
When using xterm-addon-web-links it seems that the parsing for the link isn't delimited by escape characters.

## Details
- Browser and browser version: Firefox 131.0.3
- OS version: Linux
- xterm.js version: 4.12.0, addon-web-links: 0.4.0

### Steps to reproduce

1. Print a link in the terminal immediately followed by an escape sequence (no whitespace).
2. The escape sequence isn't recognized by the rendering, causing incorrect presentation.
3. When clicking the link the escape sequence is included, mangling the link.

This issue was found in Zuul as a user, so I don't have the opportunity to upgrade or test much at all around it.

**Comment by jerch** [comment]
Thats right - weblinks is hack only operating on the terminal buffer state, thus it cannot see any terminal sequences anymore. If you want proper sequence aware link handling it is better to use OSC 8 for that and disable the web-links addon.
